osh/cmd_eval.py

```diff
@@ -1113,6 +1163,16 @@ def _Dispatch(self, node, cmd_st):
                 # TODO: What about exceptions?  They just throw?
                 status = 0
 
+            elif case(command_e.Retval):
+                node = cast(command.Retval, UP_node)
+
+                if mylib.PYTHON:
+                    val = self.expr_ev.EvalExpr(node.val, node.keyword)
+                else:
+                    # FIXME: evaluate the argument once expr_ev is translated
+                    val = value.Null
+                raise vm.ControlFlow(node.keyword, -1, val)
```

Comment:
This could be very ergonomic in ASDL. Maybe we could add a haskell-like `deriving` clause to the end of type definitions? Then we could write something like:

```asdl
control_flow =
    Break(int levels)
  | Contrinue(int levels)
  | Return(int code)
  | RetVal(value val)
  deriving [Exception]
```

But for now I'll do it manually, won't be too much work. I don't want to balloon the scope of this PR too much.